### PR TITLE
Fixes #13753

### LINKED ISSUE #13753: lambdify can't handle substituting functions and their derivatives simultaneously
It would be nice if this didn't fail:

```
moorepants@garuda:~$ ipython
Python 3.5.4 | packaged by conda-forge | (default, Dec  9 2017, 16:18:50) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.2.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import sympy as sm

In [2]: t = sm.symbols('t')

In [3]: f = sm.Function('f')(t)

In [4]: expr = f + f.diff()

In [5]: eval_expr = sm.lambdify((f, f.diff()), expr)
Traceback (most recent call last):

  File "/home/moorepants/miniconda3/lib/python3.5/site-packages/IPython/core/interactiveshell.py", line 2910, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)

  File "<ipython-input-5-423c76f9ef3a>", line 1, in <module>
    eval_expr = sm.lambdify((f, f.diff()), expr)

  File "/home/moorepants/miniconda3/lib/python3.5/site-packages/sympy/utilities/lambdify.py", line 434, in lambdify
    func = eval(lstr, namespace)

  File "<string>", line 1
    lambda _Dummy_19,Derivative(f(t), t): (Derivative(_Dummy_19, t) + _Dummy_19)
                               ^
SyntaxError: invalid syntax


In [6]: eval_expr = sm.lambdify((f.diff(), f), expr)
Traceback (most recent call last):

  File "/home/moorepants/miniconda3/lib/python3.5/site-packages/IPython/core/interactiveshell.py", line 2910, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)

  File "<ipython-input-6-7cf46a1a2b77>", line 1, in <module>
    eval_expr = sm.lambdify((f.diff(), f), expr)

  File "/home/moorepants/miniconda3/lib/python3.5/site-packages/sympy/utilities/lambdify.py", line 434, in lambdify
    func = eval(lstr, namespace)

  File "<string>", line 1
    lambda Derivative(f(t), t),_Dummy_20: (Derivative(_Dummy_20, t) + _Dummy_20)
                     ^
SyntaxError: invalid syntax
```

I think this has to do with the order that the dummies are substituted in. If `subs` is used there may be no guarantee that the order of the args passed to `lambdify` is preserved.

### PR DESCRIPTION
**Pull request by aarish17212**
Fixes #13753 

I have modified Lambdify to handle substitution functions and their derivatives simultaneously.

Now it should give outputs like this,
from sympy.abc import x
f = Function('f')(x)
f = lambdify((f,f.diff()),f+f.diff())
f(sin(x),sin(x).diff()) would give output :- sin(x) + cos(x)
f(cos(x),cos(x).diff()) would give output :- cos(x) - sin(x)

f = lambdify((f,f.diff().diff()),f+2*f.diff().diff())
f(exp(x),exp(x).diff().diff()) would give output :- 3*exp(x)

### CODE REVIEW

File: sympy/utilities/lambdify.py [hunk 1]
```diff
@@ -8,7 +8,12 @@
 from functools import wraps
 import inspect
 import textwrap
-
+from sympy import *
```
Comment (normalhuman) [inline sympy/utilities/lambdify.py]: `import *` should not be used in library code. Import what you need.

File: sympy/utilities/lambdify.py [hunk 2]
```diff
@@ -8,7 +8,12 @@
...
+# from sympy import Derivative
```
Comment (normalhuman) [inline sympy/utilities/lambdify.py]: Why is this commented-out code here?

File: sympy/utilities/lambdify.py [hunk 3]
```diff
@@ -8,7 +8,12 @@
...
+from sympy import Integer
+from sympy import diff
+from sympy import sin
+from sympy import cos
```
Comment (normalhuman) [inline sympy/utilities/lambdify.py]: Usually (look around in SymPy codebase) the import sources are more specific than that:
```
from sympy.core.numbers import Integer
from sympy.core.function import diff
from sympy.functions.elementary.trigonometric import sin, cos
```

File: sympy/utilities/lambdify.py [hunk 4]
```diff
@@ -357,14 +361,15 @@ def lambdify(args, expr, modules=None, printer=None, use_imps=True,
     for m in namespaces[::-1]:
         buf = _get_namespace(m)
         namespace.update(buf)
+    # print (namespace)
```
Comment (normalhuman) [inline sympy/utilities/lambdify.py]: Is this a leftover from debugging?

File: sympy/utilities/lambdify.py [hunk 5]
```diff
@@ -515,36 +522,51 @@ def lambdastr(args, expr, printer=None, dummify=False):
...
         if isinstance(args, str):
-            return args
+           return args
         elif isinstance(args, DeferredVector):
-            return str(args)
+           return str(args)
         elif iterable(args):
             dummies = flatten([sub_args(a, dummies_dict) for a in args])
-            return ",".join(str(a) for a in dummies)
+            return ",".join(str(a) for a in dummies if a!=None)
```
Comment (normalhuman) [inline sympy/utilities/lambdify.py]: Missing whitespace around != (see PEP8, or check with http://pep8online.com/)

File: sympy/utilities/lambdify.py [hunk 6]
```diff
@@ -515,36 +522,51 @@ def lambdastr(args, expr, printer=None, dummify=False):
...
             #Sub in dummy variables for functions or symbols
             if isinstance(args, (Function, Symbol)):
+                
                 dummies = Dummy()
                 dummies_dict.update({args : dummies})
                 return str(dummies)
+
+            elif isinstance(args,Derivative):
+                dummies = Dummy()
+                dummies_dict.update({args : dummies})
```
Comment (normalhuman) [inline sympy/utilities/lambdify.py]: I see that the whitespace before the colon was there already, but it shouldn't be. According to PEP8, it should be `{args: dummies}`.

File: sympy/utilities/lambdify.py [hunk 7]
```diff
@@ -563,6 +585,7 @@ def isiter(l):
             ','.join(dum_args), lstr, flat, iter_args)
         if len(re.findall(r'\b%s\b' % flat, rv)) > 1:
             raise ValueError('the name %s is reserved by lambdastr' % flat)
+        print (rv)
```
Comment (normalhuman) [inline sympy/utilities/lambdify.py]: Is this for debugging?

**Review by normalhuman — COMMENTED** [review]
Please follow the [development workflow](https://github.com/sympy/sympy/wiki/Development-workflow#writing-pull-request-title-and-description), in particular concerning the pull request title and description. The description should explain what you did and why you did it. 

I haven't reviewed the logic of this PR yet; waiting to see its description. 

Testing code quality locally with `bin/test quality` is very quick and would avoid several whitespace issues here.

File: sympy/utilities/lambdify.py (hunk 7 above)
Comment (ac9888185875) [reply sympy/utilities/lambdify.py]: Sorry for this inconvenience.I will make a pull request again soon after correcting all these things.

File: sympy/utilities/lambdify.py (hunk 1 above)
Comment (ac9888185875) [reply sympy/utilities/lambdify.py]: okk, it won't happen again.

File: sympy/utilities/lambdify.py (hunk 2 above)
Comment (ac9888185875) [reply sympy/utilities/lambdify.py]: It was by mistake by me, it won't happen again.

File: sympy/utilities/lambdify.py (hunk 4 above)
Comment (ac9888185875) [reply sympy/utilities/lambdify.py]: Yes

File: sympy/utilities/lambdify.py (hunk 7 above)
Comment (ac9888185875) [reply sympy/utilities/lambdify.py]: I would take care of all these things in future. I have updated now the comment. Can I get a review on my logic?

**Comment by ac9888185875** [comment]
I want to get a review on my logic @normalhuman . What should I do to make it correct?

**Comment by ac9888185875** [comment]
Why it shows an error can't import name Derivative in the travis checks @normalhuman ?

File: sympy/utilities/lambdify.py [hunk 8]
```diff
@@ -703,4 +718,4 @@ def implemented_function(symfunc, implementation):
     elif not isinstance(symfunc, UndefinedFunction):
         raise ValueError('symfunc should be either a string or'
                          ' an UndefinedFunction instance.')
-    return symfunc
+    return symfunc
```
Comment (Abdullahjavednesar) [inline sympy/utilities/lambdify.py]: Leave a line after this statement.

File: sympy/utilities/tests/test_lambdify.py [hunk 9]
```diff
@@ -5,10 +5,10 @@
 import mpmath
 from sympy.utilities.pytest import XFAIL, raises
 from sympy import (
-    symbols, lambdify, sqrt, sin, cos, tan, pi, acos, acosh, Rational,
-    Float, Matrix, Lambda, Piecewise, exp, Integral, oo, I, Abs, Function,
+    symbols, lambdify, sqrt, sin, cos, tan, pi, acos, acosh, Rational,DotProduct,
```
Comment (Abdullahjavednesar) [inline sympy/utilities/tests/test_lambdify.py]: Space before DotProduct.

File: sympy/utilities/tests/test_lambdify.py [hunk 10]
```diff
@@ -5,10 +5,10 @@
...
+    Float, Matrix, Lambda, Piecewise, exp, Integral, oo, I, Abs, Function,
```
Comment (Abdullahjavednesar) [inline sympy/utilities/tests/test_lambdify.py]: Unnecessary space

**Comment by Abdullahjavednesar** [comment]
@ac9888185875 rename the PR to something meaningful.

**Comment by czgdp1807** [comment]
Closing in reference to https://github.com/sympy/sympy/issues/13753#issuecomment-557864575
Please comment below if you think it needs to be re-opened. Thanks for the contributions. Apologies for inconvenience, if any, caused.

**Pull request closed without merging**